Plane: added TKOFF_THR_MIN

tridge opened the pull request:
this allows for a minimum throttle to be applied during a takeoff or quadplane forward transition in auto throttle modes. This is useful on high drag aircraft where normal TECS throttle control results in a slow transition

IamPete1 commented on ArduPlane/servos.cpp:
```diff
@@ -540,7 +540,21 @@ float Plane::apply_throttle_limits(float throttle_in)
+    if (use_takeoff_throttle_max) {
+        const auto tmin_thr = g2.takeoff_throttle_min;
+        if (tmin_thr > throttle) {
+            throttle = MAX(throttle, tmin_thr);
```
This means I could set my take off min throttle larger than `THR_MAX` (or `TKOFF_THR_MAX`). I think we should still constrain down in that case.

IamPete1 commented on ArduPlane/servos.cpp:
```diff
@@ -540,7 +540,21 @@ float Plane::apply_throttle_limits(float throttle_in)
+      apply TKOFF_THR_MIN if enabled and in an auto-throttle forward
+      transition
+     */
+    if (use_takeoff_throttle_max) {
```
Why not apply in the same case as `takeoff_throttle_max`. Presumably because you want to do the TECS reset. But if that is important we should probably be doing the TECS reset for the existing ICE idle, max take off throttle, battery comp and watt limit cases too....

IamPete1 replied:
also, maybe the bool should be `use_takeoff_throttle_limits` now.

IamPete1 reviewed (commented):
We could pass the actual throttle limits to TECS so it knows what is going on. I guess the reset means that it will actually chop throttle to zero and have under speed before the I term builds up.

tridge committed:
Plane: added TKOFF_THR_MIN for quadplane SLT

this allows for a minimum throttle to be applied during a forward
transition in auto throttle modes. This is useful on high drag
aircraft where normal TECS throttle control results in a slow
transition

tridge commented:
@Georacer updated to handle SLT transition

peterbarker commented:
https://github.com/ArduPilot/ardupilot/pull/27174 added this parameter and https://github.com/ArduPilot/ardupilot/pull/27770 fixed its usage.

Closing this one now.
